R/lazy-join-query.R

```diff
@@ -156,15 +201,83 @@ sql_build.lazy_join_query <- function(op, con, ...) {
   )
 }
 
+#' @export
+sql_build.lazy_multi_join_query <- function(op, con, ...) {
+  auto_name <- is.na(op$table_names$as)
+  table_names_out <- dplyr::coalesce(op$table_names$as, op$table_names$name)
+  if (length(table_names_out) == 2) {
+    table_names_out <- join_simple_table_alias(op$table_names$name, op$table_names$as)
+
+    if (op$joins$type %in% c("full", "right")) {
+      # construct a classical `join_query()` so that special handling of
```

Comment:
I think `inner_join()`, `left_join()` and `cross_join()` should work the same in every database. But I think this fallback makes sense for `right_join()` and `full_join()`.